### PR TITLE
folding@home: 6.02 -> 7.5.1

### PR DESCRIPTION
**Pull request by zimbatm**
###### Motivation for this change

[Find a cure to Covid-19](https://foldingathome.org/2020/02/27/foldinghome-takes-up-the-fight-against-covid-19-2019-ncov/)

Update folding@home. Things have changed quite a bit so I rebuilt everything from scratch.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zimbatm** [comment]
currently debugging some folding issues:
```
15:55:14:WU01:FS00:Starting
15:55:14:WARNING:WU01:FS00:AS lowered CPUs from 15 to 14
15:55:14:WU01:FS00:Running FahCore: /nix/store/arcgrgdakzhwll6ngd2k6smjjp698j6z-fahclient-7.5.1/bin/FAHCoreWrapper /var/lib/private/foldingathome/cores/cores.foldingathome.org/v7/lin/64bit/avx/Core_a7.fah/FahCore_a7 -dir 01 -suffix 01 -version 705 -lifeline 4043 -checkpoint 15 -np 14
15:55:14:WU01:FS00:Started FahCore on PID 6029
15:55:14:WU01:FS00:Core PID:6033
15:55:14:WU01:FS00:FahCore 0xa7 started
15:55:15:WARNING:WU01:FS00:FahCore returned: FAILED_3 (255 = 0xff)
```

**Comment by zimbatm** [comment]
## How to test the PR

1. Change your NIX_PATH to point to this PR.
2. Add the following entries in your NixOS configuration:
```nix
{ pkgs, ... }:
{
  services.foldingathome.enable = true;
  environment.systemPackages = with pkgs; [ fahviewer fahcontrol ];
}
```
3. `nixos-rebuild switch`
4. Run `FAHControl` and look at the logs. You might also want to configure your Team ID to 236565 :)

**Comment by zimbatm** [comment]
image

**Comment by Infinisil** [comment]
Partial duplicate of #81768

**Comment by rissson** [comment]
You might also want to update the nixos module. It seems quite out of date

**Comment by zimbatm** [comment]
@rissson what do you mean, I rewrote it from scratch? :D

File: pkgs/applications/science/misc/foldingathome/client.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    inherit version;
+    pname = "fahclient";
+
+    src =
+      if stdenv.hostPlatform.system == "x86_64-linux" then
+        fetchurl {
+          url = "https://download.foldingathome.org/releases/public/release/fahclient/debian-stable-64bit/v${majMin}/fahclient_${version}_amd64.deb";
+          hash = "sha256-7+RwYdMoZnJZwYFbmLxsN9ozk2P7jpOGZz9qlvTTfSY=";
+        }
+      else throw "${pname}-${version} is not supported on ${stdenv.hostPlatform.system}";
```
[the remark this answers is not in the document]
Comment (zimbatm) [reply pkgs/applications/science/misc/foldingathome/client.nix]: yeah good point

**Comment by rissson** [comment]
@zimbatm yes indeed I was looking at the old one. It would be nice to also have an option for the config.xml file

**Comment by zimbatm** [comment]
@rissson: now the FAHClient can be started with extra options and defaults to the NixOS Team ID. I think this approach is more reliable than providing the config.xml since the service expects the file to be mutable.

**Comment by jcumming** [comment]
Thanks! I've started this on a few boxes now. :)

File: pkgs/applications/science/misc/foldingathome/viewer.nix [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+, fetchurl
+, freeglut
+, gcc-unwrapped
+, libGL
+, libGLU
+, makeWrapper
+, zlib
+}:
+let
+  majMin = "7.5";
```
Comment (jonringer) [inline pkgs/applications/science/misc/foldingathome/viewer.nix]: for consistency
```suggestion
  majMin = stdenv.lib.versions.majorMinor version;
```

File: pkgs/applications/science/misc/foldingathome/viewer.nix [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  inherit version;
+  pname = "fahviewer";
+
+  src =
+    if stdenv.hostPlatform.system == "x86_64-linux" then
+      fetchurl {
+        url = "https://download.foldingathome.org/releases/public/release/fahviewer/debian-stable-64bit/v${majMin}/fahviewer_${version}_amd64.deb";
+        hash = "sha256-yH0zGjX8aNBEJ5lq7wWydcpp2rO+9Ah++q9eJ+ldeyk=";
+      }
+    else throw "${pname}-${version} is not supported on ${stdenv.hostPlatform.system}";
```
Comment (jonringer) [inline pkgs/applications/science/misc/foldingathome/viewer.nix]: `meta.platforms = [ "x86_64-linux" ];` is already set
```suggestion
      fetchurl {
        url = "https://download.foldingathome.org/releases/public/release/fahviewer/debian-stable-64bit/v${majMin}/fahviewer_${version}_amd64.deb";
        hash = "sha256-yH0zGjX8aNBEJ5lq7wWydcpp2rO+9Ah++q9eJ+ldeyk=";
      };
```

File: pkgs/applications/science/misc/foldingathome/control.nix [hunk 4]
```diff
@@ -0,0 +1,60 @@
+{ stdenv
+, autoPatchelfHook
+, dpkg
+, fetchurl
+, makeWrapper
+, python2
+}:
+let
+  majMin = "7.5";
```
Comment (jonringer) [inline pkgs/applications/science/misc/foldingathome/control.nix]:
```suggestion
  majMin = stdenv.lib.versions.majorMinor version;
```

File: pkgs/applications/science/misc/foldingathome/control.nix [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  inherit version;
+  pname = "fahcontrol";
+
+  src =
+    if stdenv.hostPlatform.system == "x86_64-linux" then
+      fetchurl {
+        url = "https://download.foldingathome.org/releases/public/release/fahcontrol/debian-stable-64bit/v${majMin}/fahcontrol_${version}-1_all.deb";
+        hash = "sha256-ydN4I6vmZpI9kD+/TXxgWc+AQqIIlUvABEycWmY1tNg=";
+      }
+    else throw "${pname}-${version} is not supported on ${stdenv.hostPlatform.system}";
```
Comment (jonringer) [inline pkgs/applications/science/misc/foldingathome/control.nix]:
```suggestion
  src = fetchurl {
        url = "https://download.foldingathome.org/releases/public/release/fahcontrol/debian-stable-64bit/v${majMin}/fahcontrol_${version}-1_all.deb";
        hash = "sha256-ydN4I6vmZpI9kD+/TXxgWc+AQqIIlUvABEycWmY1tNg=";
      };
```

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

**Comment by rissson** [comment]
> @rissson: now the FAHClient can be started with extra options and defaults to the NixOS Team ID. I think this approach is more reliable than providing the config.xml since the service expects the file to be mutable.

So you recommend modifying the config with FAHControl then?

**Comment by zimbatm** [comment]
> So you recommend modifying the config with FAHControl then?
FAHClient should configure the xml when started with the right arguments

**Comment by zimbatm** [comment]
Ok I think this is ready. Rebased and squashed. I also added the NixOS module renamed options for back-compat.

**Comment by olejorgenb** [comment]
Maybe add support for passkeys also? https://foldingathome.org/support/faq/points/passkey/

**Comment by zimbatm** [comment]
I'd rather get this merged. The passkey support doesn't seem strictly necessary to me and can be added using FAHControl.

**Review by jonringer — APPROVED** [review]
diff LGTM
programs seem to work

```
[13 built, 198 copied (1226.2 MiB), 235.5 MiB DL]
https://github.com/NixOS/nixpkgs/pull/82192
1 package blacklisted:
tests.nixos-functions.nixos-test

3 package built:
fahclient fahcontrol fahviewer
```

**Pull request merged**